pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -456,106 +452,73 @@ func (nim *nodeInfoManager) installDriverToCSINodeInfo(
-		Available: true,
-		// TODO(https://github.com/kubernetes/enhancements/issues/625): Add actual migration status
-		VolumePluginMechanism: csiv1alpha1.VolumePluginMechanismInTree,
-	}
 
 	newDriverSpecs = append(newDriverSpecs, driverSpec)
-	newDriverStatuses = append(newDriverStatuses, driverStatus)
 	nodeInfo.Spec.Drivers = newDriverSpecs
-	nodeInfo.Status.Drivers = newDriverStatuses
 
-	err := validateCSINodeInfo(nodeInfo)
-	if err != nil {
-		return err
-	}
-	_, err = csiKubeClient.CsiV1alpha1().CSINodeInfos().Update(nodeInfo)
+	_, err := csiKubeClient.StorageV1beta1().CSINodes().Update(nodeInfo)
 	return err
 }
 
-func (nim *nodeInfoManager) uninstallDriverFromCSINodeInfo(
+func (nim *nodeInfoManager) uninstallDriverFromCSINode(
 	csiDriverName string) error {
 
-	csiKubeClient := nim.volumeHost.GetCSIClient()
+	csiKubeClient := nim.volumeHost.GetKubeClient()
 	if csiKubeClient == nil {
 		return fmt.Errorf("error getting CSI client")
 	}
 
 	var updateErrs []error
 	err := wait.ExponentialBackoff(updateBackoff, func() (bool, error) {
-		if err := nim.tryUninstallDriverFromCSINodeInfo(csiKubeClient, csiDriverName); err != nil {
+		if err := nim.tryUninstallDriverFromCSINode(csiKubeClient, csiDriverName); err != nil {
 			updateErrs = append(updateErrs, err)
 			return false, nil
 		}
 		return true, nil
 	})
 	if err != nil {
-		return fmt.Errorf("error updating CSINodeInfo: %v; caused by: %v", err, utilerrors.NewAggregate(updateErrs))
+		return fmt.Errorf("error updating CSINode: %v; caused by: %v", err, utilerrors.NewAggregate(updateErrs))
 	}
 	return nil
 }
 
-func (nim *nodeInfoManager) tryUninstallDriverFromCSINodeInfo(
-	csiKubeClient csiclientset.Interface,
+func (nim *nodeInfoManager) tryUninstallDriverFromCSINode(
+	csiKubeClient clientset.Interface,
 	csiDriverName string) error {
 
-	nodeInfoClient := csiKubeClient.CsiV1alpha1().CSINodeInfos()
+	nodeInfoClient := csiKubeClient.StorageV1beta1().CSINodes()
 	nodeInfo, err := nodeInfoClient.Get(string(nim.nodeName), metav1.GetOptions{})
-	if err != nil {
-		return err // do not wrap error
+	if err != nil && errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return err
 	}
 
-	hasModified := false
-	newDriverStatuses := []csiv1alpha1.CSIDriverInfoStatus{}
-	for _, driverStatus := range nodeInfo.Status.Drivers {
-		if driverStatus.Name == csiDriverName {
-			// Uninstall the driver if we find it
-			hasModified = driverStatus.Available
-			driverStatus.Available = false
+	// Uninstall matching CSINodeDriver
+	for i, driver := range nodeInfo.Spec.Drivers {
+		if driver.Name == csiDriverName {
+			nodeInfo.Spec.Drivers = append(nodeInfo.Spec.Drivers[:i], nodeInfo.Spec.Drivers[i+1:]...)
 		}
-		newDriverStatuses = append(newDriverStatuses, driverStatus)
 	}
 
-	nodeInfo.Status.Drivers = newDriverStatuses
-
-	if !hasModified {
-		// No changes, don't update
-		return nil
+	if len(nodeInfo.Spec.Drivers) == 0 {
+		err = nodeInfoClient.Delete(string(nim.nodeName), nil)
```

Comment:
please remove all/any code that deletes the `CSINodeInfo` object as a whole. We still want that to hang around even without any drivers because of the annotation we will be adding for CSI Migration

Reply:
Ok